### PR TITLE
Add Nanocl in Self-Hosting Solution

### PR DESCRIPTION
**Pull request by leon3s**
Thanks for taking the time to suggest an addition to awesome-selfhosted! 

To ensure your Pull Request is dealt with swiftly, please check the following (check the boxes `[x]`):
- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted-data/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pulls), including closed ones.
- [x] Any software you are adding is not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] The file you are adding is formatted as described in [addition.md](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/ISSUE_TEMPLATE/addition.md).
- [x] `Demo` links should only be used for interactive demos, i.e. not video demonstrations. 
- [x] Comments and unused optional fields have been removed.
- [x] The file you are adding uses [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) file naming, for example `my-awesome-software.yml`.
- [x] Values for `platform` are the main **server-side** requirements for the software. Don't include frameworks or specific dialects.
- [x] Any software project you are adding to the list is actively maintained.
- [x] Any software project you are adding was first released more than 4 months ago.
- [x] Any software project you are adding has working installation instructions.
- [x] If you are adding software forked from another active project, please provide/link to a clear list of differences between both.
- [x] You understand that your Pull Request will be merged ~1 week after approval, to allow for further comments if needed.

### CODE REVIEW

File: software/nanocl.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Nanocl
+website_url: https://next-hat.com/nanocl
+description: A developer friendly kubernetes alternative
```
Comment (KieranRobson) [inline software/nanocl.yml:3]:
```suggestion
description: Distributed System that simplifies the management of containers and virtual machines (alternative to Kubernetes). 
```

File: software/nanocl.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+licenses:
+  - Apache-2.0
+  - MIT
+platforms:
+  - Docker
+  - Deb
+  - Rust
```
Comment (KieranRobson) [inline software/nanocl.yml:10]:
```suggestion
platforms:
  - Docker
  - deb
  - Rust
```

**Review by KieranRobson — CHANGES_REQUESTED** [review]
Couple changes to be made.

**Comment by leon3s** [comment]
Thanks for the review, it have been updated!

**Comment by nodiscc** [comment]
> If it doesn't fit then why Casaos, DockSTARTer and xsrv got accepted ? They also came with a CLI.

The main difference is, I think, that these projects come with premade installation routines for self-hosted software (media servers, communication services, ...) in an "appstore"-like fashion. These projects _rely on_ orchestration/deployment automation tools similar to Nanocl - but you will notice that we don't list `vagrant`, `ansible`, `proxomox`, `virsh` and other config management/generic provisioning/lab management solutions, as these fall into the more generic realm of system administration tools (even when they do provide APIs) - these tools are listed at https://github.com/awesome-foss/awesome-sysadmin or https://landscape.cncf.io/?license=open-source.

I agree the distinction is thin, and we could improve discoverability of related lists using _See also:_ links like we already do on some pages - but the line has to be drawn _somewhere_, at the risk of being swamped into generic admin tools, which is not the goal of this list.

What do you think?

**Comment by leon3s** [comment]
You mentioned:
> This is a list of [Free](https://en.wikipedia.org/wiki/Free_software) Software [network services](https://en.wikipedia.org/wiki/Network_service) and [web applications](https://en.wikipedia.org/wiki/Web_application) which can be hosted on your own server(s).

On the network wikipedia page they give the example of a DNS server.
Nanocl helps you manage your own dns server so I really don't see why it doesn't fit the description.

I think it's nice that you mention the "appstore" because one of the goals of Nanocl is to have reusable yaml file where you can pass argument to allow better flexibility.
We have an appstore, it just doesn't come with a fancy UI and a lot of preset services, because my project is less mature than the others.
I invite you to check my [vpn guide](https://docs.next-hat.com/guides/nanocl/advanced-usage/vpn) as an example.
We have a registery of ready to deploy services that can be found [there](https://nhnr.io/)
We also have a dashboard if it's the UI the problem, it's just doesn't have all the feature the CLI has.

For me Nanocl is a selfhosted service that helps you to selfhost.
This is the description of the `Self-Hosting` Solution category:
> Software for easy installation, management and configuration of self-hosted services and applications.

And thats exactly what Nanocl does so I'm really confused.

Also I wont define vagrant, promox and virsh as sysadmin tools but that is my opinion.

**Pull request closed without merging**